src/compiler/checker.ts

```diff
@@ -4504,6 +4535,12 @@ namespace ts {
                 if (!popTypeResolution()) {
                     type = reportCircularityError(symbol);
                 }
+
+                // If the type is a fresh, unique symbol type and is not the type for this symbol, get its regular type
+                if (type.flags & TypeFlags.Unique && type.flags & TypeFlags.Fresh && type.symbol !== symbol) {
+                    type = (<UniqueType>type).regularType;

```

Comment:
the regular type for a unique type is usually just Symbol, right? I don't understand why these two lines are here.